compiler/daml-lf-ast/src/DA/Daml/LF/TemplateOrInterface.hs

```diff
@@ -0,0 +1,15 @@
+-- Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module DA.Daml.LF.TemplateOrInterface
```

Comment:
I do not really know Haskell convention.  But it is really worthwhile to create a new file for that ? Putting it in `Ast` or `Util` would not be good enought ?

Reply:
It doesn't _have_ to be in a new file, but in either of those we would get clashes with [`data Template`](https://github.com/digital-asset/daml/blob/main/compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Base.hs#L907). Modules using both `Ast` and this module unqualified would still get clashes, since in Haskell we can't namespace the constructors in an `object` like we do in Scala. The best we can do is import this module qualified, 

```haskell
import qualified DA.Daml.LF.TemplateOrInterface as TemplateOrInterface
```

this way we get the same syntax as in Scala, `TemplateOrInterface.Template`

Reply:
fair enough